lua/entities/gmod_wire_expression2/core/custom/cfc_e2_lib.lua

```diff
@@ -137,3 +137,46 @@ e2function number array:indexOf(vector4 vec4)
     return indexOfPackedValue( this, vec4 )
 end
 
+
+-- String Functions
+e2function number string:isNumber()
+    -- This regex tests for only digits
+    if string.find( this, "^%d*$" ) then return 1 end
```

Comment:
How about `tonumber (e [, base])`?
"Tries to convert its argument to a number. If the argument is already a number or a string convertible to a number, then tonumber returns this number; otherwise, it returns nil."

```lua
e2function number string:isNumber()
   if tonumber( this ) then return 1 end
   return 0
end
```
I would like to just do: `return tonumber(this)`, but if it's zero, it will return 0, which E2 considers _false_ and lua considers _true_!